src/de.rs

```diff
@@ -21,6 +25,53 @@ impl<'de> Deserialize<'de> for std::time::Duration {
     }
 }
 
+impl<'de> Deserialize<'de> for Option<std::time::Duration> {
```

Comment:
So, just adding the standard "looks like you could macro this if you wanted to cut code count" comment. Presumably not really worth it for 50 LoC and a single duplication.

Reply:
I would stand by readibility > single duplication in this case 🙂 .

Reply:
Too little duplication to warrant a macro imo.